openqaoa/optimizers/result.py

```diff
@@ -62,30 +62,30 @@ def __init__(
         self.cost_hamiltonian = cost_hamiltonian
 
         self.evals = {
-            "number of evals": log.func_evals.best[0],
-            "jac evals": log.jac_func_evals.best[0],
-            "qfim evals": log.qfim_func_evals.best[0],
+            "number_of_evals": log.func_evals.best[0],
+            "jac_evals": log.jac_func_evals.best[0],
+            "qfim_evals": log.qfim_func_evals.best[0],
         }
 
         self.intermediate = {
-            'angles log': np.array(log.param_log.history).tolist(),
-            'intermediate cost': log.cost.history,
-            'intermediate measurement outcomes':
+            'angles_log': np.array(log.param_log.history).tolist(),
```

Comment:
Shall we rename this to `intermediate_angles`? @vishal-ph

Reply:
@Q-lds , @raulconchello. So one point about naming these fields. Since the outer field is already called intermediate, maybe we should rename all of them to just "cost", "measurement_outcomes", "angles" (getting rid of the word intermediate)

Similarly for the field optimized, lets remove the optimized from the sub fields as well. It's redundant in my opinion.

Reply:
It is redundant, but maybe they can be useful at some point. For example, when you are looking at a json of the results, the file is very long, so it can be helpful if the keys are "intermediate_..." or "optimized_...", so you know what you are looking at without the need of scrolling up to the parent. 
I think it doesn't hurt.

Reply:
These fields are supposed to be accessed programmatically so I am in favor of what Vishal is saying. 

I also have another question: how is the optimized set chosen? The lowest value in the optimization? Or the last? If it is the last, then the true cost-history will be `intermediate.cost` + `optimized.cost` right? 

I think we need to be a bit more explicit with this: how about adding what optimisation round does the optimized result corresponds to?

Reply:
@shahidee44 will respond better than I can. 
Can we include the optimization round?

Reply:
> @Q-lds , @raulconchello. So one point about naming these fields. Since the outer field is already called intermediate, maybe we should rename all of them to just "cost", "measurement_outcomes", "angles_log" (getting rid of the word intermediate)
> 
> Similarly for the field optimized, lets remove the optimized from the sub fields as well. It's redundant in my opinion.

Ok, I have updated the keys to be the same for both dictionaries `intermediate` and `optimized`:
```
        self.intermediate = {
            'angles': ...,
            'cost': ...,
            'measurement_outcomes': ...,
            'job_id': ...,
        }

        self.optimized = {
            'angles': ...,
            'cost': ...,
            'measurement_outcomes': ...,
            'job_id': ...,
        }
```

Reply:
> These fields are supposed to be accessed programmatically so I am in favor of what Vishal is saying.
> 
> I also have another question: how is the optimized set chosen? The lowest value in the optimization? Or the last? If it is the last, then the true cost-history will be `intermediate.cost` + `optimized.cost` right?
> 
> I think we need to be a bit more explicit with this: how about adding what optimisation round does the optimized result corresponds to?

@shahidee44 will respond better than I can. 
Can we include the optimization round?

Reply:
In the optimized result, yes. I think so!

Reply:
Yes we can. Although, i'd be very careful about what we call that value. We can always repurpose the evaluations logger variable for such a task. Or just create a new variable for it if its not possible.

Reply:
I have done the following to have `q.results.optimized['eval_number']` (an int which tells the evaluation in which we have found the optimized values)
https://github.com/entropicalabs/openqaoa/blob/a8e14d9d0f3e9d2dcfe555a682817c5fcbad0e01/openqaoa/optimizers/training_vqa.py#L230

Reply:
@vishal-ph ?

Reply:
Now that the functionality is already added, lets keep it.

Reply:
@raulconchello, you can resolve this thread.